articles/css-flexbox-vs-css-grid/index.md

```diff
@@ -0,0 +1,767 @@
+
+Let's see how to implement that.
+
+Here I have nine items inside a Flexbox container.
+
+```html
+<div class="gap">
+    <div>A</div>
+    <div>B</div>
+    <div>C</div>
+    <div>D</div>
+    <div>E</div>
+    <div>F</div>
+    <div>G</div>
+    <div>H</div>
+    <div>I</div>
+</div>
+```
+
+```css
+.gap {
+    display: flex;
+    flex-wrap: wrap;
+}
+
+.gap>div {
+    background-color: rgba(10, 129, 209, .5);
+    border: 1px solid #000;
+    color: #000;
+    padding: 20px;
+    font-size: 150%;
+}
+```
+
+![](/engineering-education/css-flexbox-vs-css-grid/gap.png)
+
+We are using margin to create gutter around the items.
+
+```css
+.gap {
+    display: flex;
+    border: 3px solid rgba(6, 109, 243, 0.911);
+    flex-wrap: wrap;
+    max-width: 800px;
+}
+
+.gap>div {
+    background-color: rgba(10, 129, 209, .5);
+    border: 1px solid #000;
+    color: #000;
+    padding: 20px;
+    font-size: 150%;
+    margin: 6px;
+}
+```
+
+![](/engineering-education/css-flexbox-vs-css-grid/gap-with-margin.png)
+
+![](/engineering-education/css-flexbox-vs-css-grid/gap-with-margin-wrapped.png)
+
+When I introduce a `margin of 6px`, notice how the spaces are created around the items. Margin creates some extra white spaces that can be avoided using the gap property.
+
+The gap property distinctly creates gutter. You can choose to create spaces between rows and columns.
+
+Apply gap by specifying `column-gap`.
+
+```css
+.gap {
+    display: flex;
+    border: 3px solid rgba(6, 109, 243, 0.911);
+    flex-wrap: wrap;
+    column-gap: 6px;
+    max-width: 800px;
+}
+
+.gap>div {
+    background-color: rgba(10, 129, 209, .5);
+    border: 1px solid #000;
+    color: #000;
+    padding: 20px;
+    font-size: 150%;
+}
+```
+
+![](/engineering-education/css-flexbox-vs-css-grid/column-gap.png)
+
+![](/engineering-education/css-flexbox-vs-css-grid/column-gap-wrapped.png)
+
+When we apply the wrap, we force Flexbox to create rows and columns on small screens. We can use `row-gap` to create gutter around them too.
+
+```css
+.gap {
+    display: flex;
+    border: 3px solid rgba(6, 109, 243, 0.911);
+    flex-wrap: wrap;
+    column-gap: 6px;
+    row-gap: 6px;
+    max-width: 800px;
+}
+
+.gap>div {
+    background-color: rgba(10, 129, 209, .5);
+    border: 1px solid #000;
+    color: #000;
+    padding: 20px;
+    font-size: 150%;
+}
+```
+
+![](/engineering-education/css-flexbox-vs-css-grid/column-gap-row-gap.png)
+
+**The shorthand**
+
+To create spacing between rows and columns, we can use gaps and specify the units of both. For example, `gap: 6px 6px` and we will get the same results as applying `column-gap: 6px` and `row-gap: 6px`.
+
+![](/engineering-education/css-flexbox-vs-css-grid/column-gap-row-gap.png)
```

Comment:
@kimkimani, I couldn't replicate this image with the above code. The grid items are still all in one row instead of in a 3x3 grid. Is this the expected behaviour?